Update Prometheus Agent doc (#17591)
prometheus/prometheus: docs/prometheus_agent.md

In essence, it looks like this:
![Prometheus Agent Remote Write](./images/prometheus_agent.png)

### Benefits of agent mode

> - Improved efficency. The customized Agent TSDB WAL removes the data immediately after successful writes. If it cannot reach the remote endpoint, it persists the data temporarily on the disk until the remote endpoint is back online. This is currently limited to a two-hour buffer only, similar to non-agent Prometheus. This means that we don't need to build chunks of data in memory. We don't need to maintain a full index for querying purposes. Essentially the Agent mode uses a fraction of the resources that a normal Prometheus server would use in a similar situation.

Reviewer: ```suggestion
- Improved efficency. The customized Agent TSDB WAL removes the data immediately after successful writes. If it cannot reach the remote endpoint, it persists the data temporarily on the disk until the remote endpoint is back online. This is currently limited to a two-hour buffer only, similar to non-agent Prometheus. This means no need to build chunks of data in memory and no need to maintain a full index for querying purposes. Essentially the Agent mode uses a fraction of the resources that a normal Prometheus server would use in a similar situation.
```
Nit: "We" aren't doing anything here.

Author: Thanks, good catch in the original wording.
